ql/src/test/queries/clientpositive/explain_big_table_multiple_ptf.q

```diff
@@ -0,0 +1,31 @@
+create table explain_multiple_ptf_big_table
+(
+    key1       string,
+    value_str1 string,
+    key3       string
+);
+create table explain_multiple_ptf_big_table2
+(
+    key21       string,
+    value_str21 string,
+    key23       string
+);
+alter table explain_multiple_ptf_big_table
+    update statistics set('numRows' = '4611686036854775807',
+                          'rawDataSize' = '922337203685477500');
+alter table explain_multiple_ptf_big_table2
+    update statistics set('numRows' = '4611686036854775807',
+                          'rawDataSize' = '9223372036854775800');
+explain
+select *,row_number() over (partition by key order by key2 desc) rn,
+       row_number() over (partition by key2 order by key desc) rn2,
+       max(value_str) over (partition by key2 order by key desc) max1,
+       max(value_str) over (partition by key order by key2 desc) max3,
+       min(value_str) over (partition by key2 order by key desc) min1,
+       min(value_str) over (partition by key order by key2 desc) min3,
+       last_value(value_str) over (partition by key) lv,
+       first_value(value_str) over (partition by key2) fv,
+       max(value_str) over (partition by key) fv21
+from (select key1 key, value_str1 value_str, key3 key2 from explain_multiple_ptf_big_table
+union all select key21 key, value_str21 value_str, key23  key2 from  explain_multiple_ptf_big_table2) a1
+;
```

Comment:
> NIT: Please add a line break if possible.
@okumin Hello, I do not have permission to access the two links you provided, so I cannot see the specific information. Regarding the "add a line" issue, I used the DG tool to format the SQL, but I am not sure if it meets the requirements.

Reply:
You can sign in Jenkins with your GitHub account, and you can see the result of CI. I will also document it.
It is just a regulation of not Hive but POSIX to end a file with \n. Thanks for tweaking it

Reply:
> You can sign in Jenkins with your GitHub account, and you can see the result of CI. I will also document it. It is just a regulation of not Hive but POSIX to end a file with \n. Thanks for tweaking it

Thank you, I think I understand now. There are still some minor formatting issues. I will run it again. I'm not sure if it's because of my local computer's performance, but each test run takes more than 20 minutes.

Reply:
After fixing the SQL formatting, the generated out file shows no significant difference.

Reply:
`numRows=9223372036854775` seems like the one of the old revision. I'd check `ql/src/test/queries/clientpositive/explain_big_table_multiple_ptf.q` has the expected content, which means not `update statistics set('numRows' = '9223372036854775'` but `update statistics set('numRows' = '4611686036854775807'`, in the test environment, and then would regenerate `ql/src/test/results/clientpositive/llap/explain_big_table_multiple_ptf.q.out`. As long as `ql/src/test/results/clientpositive/llap/explain_big_table_multiple_ptf.q.out` retains `update statistics set('numRows' = '9223372036854775'`, we will have no luck on Jenkins.

```
[2024-12-08T12:12:37.765Z] <     update statistics set('numRows' = '4611686036854775807',
[2024-12-08T12:12:37.765Z] <                           'rawDataSize' = '922337203685477500')
[2024-12-08T12:12:37.765Z] ---
[2024-12-08T12:12:37.765Z] >     update statistics set('numRows' = '9223372036854775',
[2024-12-08T12:12:37.765Z] >                           'rawDataSize' = '9223372036854775807')
```

Reply:
@okumin  Should I change the .q file name to avoid this issue? Although this is not a reasonable solution.

Reply:
I think I have identified the cause of the issue. It was due to a problem with my local environment. I should first run:

mvn clean install -DskipTests
at the root directory, then go to the itests directory and run:

cd itests
mvn clean install -DskipTests

Finally, I should execute:

mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=explain_multiple_ptf_on_big_table.q -Dtest.output.overwrite=true
instead of directly running the test after making code changes. Skipping the build steps caused the explain_multiple_ptf_on_big_table.q.out file to remain the old version rather than being updated to reflect the latest changes.

Reply:
Sorry, I will make the changes. I wasted your time.

Reply:
No problem at all. I am aware that it is tricky how qtest works.

The last thing is, maybe some of your scripts removed extra spaces when you committed the change. `qtest` requires any bytes equal to the expected bytes, which means we have to keep any characters on `*.q.out` as it is.

```
[2024-12-08T19:16:46.814Z] <         Map 1 
[2024-12-08T19:16:46.814Z] ---
[2024-12-08T19:16:46.814Z] >         Map 1
[2024-12-08T19:16:46.814Z] 137c137
[2024-12-08T19:16:46.814Z] <         Map 8 
[2024-12-08T19:16:46.814Z] ---
[2024-12-08T19:16:46.814Z] >         Map 8
[2024-12-08T19:16:46.814Z] 155c155
[2024-12-08T19:16:46.814Z] <         Reducer 3 
[2024-12-08T19:16:46.814Z] ---
[2024-12-08T19:16:46.814Z] >         Reducer 3
[2024-12-08T19:16:46.814Z] 205c205
[2024-12-08T19:16:46.814Z] <         Reducer 4 
[2024-12-08T19:16:46.814Z] ---
[2024-12-08T19:16:46.814Z] >         Reducer 4
[2024-12-08T19:16:46.814Z] 255c255
[2024-12-08T19:16:46.814Z] <         Reducer 5 
[2024-12-08T19:16:46.814Z] ---
[2024-12-08T19:16:46.814Z] >         Reducer 5
[2024-12-08T19:16:46.814Z] 293c293
[2024-12-08T19:16:46.814Z] <         Reducer 6 
[2024-12-08T19:16:46.814Z] ---
[2024-12-08T19:16:46.814Z] >         Reducer 6
[2024-12-08T19:16:46.814Z] 331c331
[2024-12-08T19:16:46.814Z] <         Reducer 7 
[2024-12-08T19:16:46.814Z] ---
[2024-12-08T19:16:46.814Z] >         Reducer 7
[2024-12-08T19:16:46.814Z] 369c369
[2024-12-08T19:16:46.814Z] <         Union 2 
[2024-12-08T19:16:46.814Z] ---
[2024-12-08T19:16:46.814Z] >         Union 2
```

If it is hard to resolve, you may edit the `q.out` by hand and add spaces. You can test it in your environment using  `mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=explain_multiple_ptf_on_big_table.q`. Without `-Dtest.output.overwrite=true`, it works in the same way of Jenkins. That means if the actual result is inconsistent with `explain_multiple_ptf_on_big_table.q.out`, it fails.

My recommended steps.
1. Run `mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=explain_multiple_ptf_on_big_table.q` in the current revision and confirms that it surely fails. If it succeeds, maybe you are executing the command in a wrong way
2. Edit `ql/src/test/results/clientpositive/llap/explain_big_table_multiple_ptf.q.out` by hand, or rerun `mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=explain_multiple_ptf_on_big_table.q -Dtest.output.overwrite=true`
3. Commit and finalize the change before something updates `ql/src/test/results/clientpositive/llap/explain_big_table_multiple_ptf.q.out`
4. Run `mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=explain_multiple_ptf_on_big_table.q` again and confirm it succeeds

Reply:
Thank you for providing the information. I will follow these steps and test it tonight.